src/ngraph/graph_util.hpp

```diff
@@ -218,6 +218,14 @@ namespace ngraph
         return result_list;
     }
 
+    template <typename T>
+    bool is_dynamic(const T& dimensions)
```

Comment:
It really only makes sense to define this on `PartialShape`. A `Shape` is never dynamic, by definition.